components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+define(['require', 'lodash', 'jquery', 'log', 'cronstrue', 'jquery_timepicker'],
+    function (require, _ , $, log, Cronstrue, Timepicker) {
+
+        var CronGenerator = function(){
+            var self = this;
+            this._cronGenerator = $('#cronGenerator').clone();
+        };
+        var errorMsg;
+
+        var isValidTimeValue = function(time, val) {
+            if(time === '*') {
+                return true;
+            }
+            if (time.search(/[^\d-,*\/]/) !== -1) {
+                return false;
+            }
+            var list = time.split(",");
+            return list.every(function (time){
+                if(time.includes('/')) {
+                    var startingTimeOptionArr = time.split('/');
+                    return isValidateTime(startingTimeOptionArr, val) ||
+                            (startingTimeOptionArr[0] === '*' && isValidateTime([startingTimeOptionArr[1]], val));
+                } else if(time.includes('-')) {
+                    var timeRangeArr = time.split('-');
+                    return isValidateTime(timeRangeArr, val) && (parseInt(timeRangeArr[0]) < parseInt(timeRangeArr[1]));
+                } else if (parseInt(time) >= 0 && parseInt(time) <= val) {
+                    return true;
+                } else {
+                    return false;
+                }
+            });
+        };
+
+        var isValidateTime = function(dataArray, value) {
+            return dataArray.every(element => {
+                if(parseInt(element) < 0 || parseInt(element) > value){
+                    errorMsg = "Numeric values between " + 0 + " and " + value;
+                } else {
+                    errorMsg = "Invalid cron Expression";
+                }
+                return parseInt(element) >= 0  && parseInt(element) <= value;
+            });
+        };
+
+        var isValidateNumericValue = function(monthArr, val, endVal) {
+            return monthArr.every(month => {
+                if(parseInt(month) < val || parseInt(month) > endVal){
+                    errorMsg = "Numeric values between " + val + " and " + endVal;
+                } else {
+                    errorMsg = "Invalid Cron Expression"
+                }
+                return parseInt(month) >= val && parseInt(month) <= endVal;
+            });
+        };
+
+        var isValidDayOfMonthValue = function(dayOfMonth, dayOfWeek) {
+            if((dayOfMonth === '*' && dayOfWeek !== '*') || (dayOfMonth === '?' && dayOfWeek !== '?')) {
```

Comment:
You do not need to check for valid cases . validate method only need to return false for wrong cases

Reply:
fixed in 0fb87f3